src/PID.c

```diff
@@ -9,14 +9,14 @@ const struct PID DEFAULT_PID = { 5, 3, 3 };
 /**
  Gets the output of the input passed through the PID controller for the delta time interval
  */
-int pid(int input, int *errorSum, const unsigned long * const deltaTime, int target, PID pid) {
+float pid(float input, double *errorSum, float prevError, const unsigned long * const deltaTime, float target, const PID pid) {
     // Generate error signal
-	int error = target - input;
-	*errorSum += error;
-	int dError = error - *errorSum;
+	float error = target - input;
+    *errorSum += error * *deltaTime;
```

Comment:
While I'm alright with the use of soft tabs, the rest of this file uses hard tabs. The rest of the tabs in this file should probably be replaced with 4 tabs.